main.cpp

```diff
@@ -1,42 +1,107 @@
 #include "main.h"
 #include "rc.h"
 
+int board_width = 8;
+int board_height = 8;
+COLORREF color1 = RGB(255, 255, 0); 
+COLORREF color2 = RGB(255, 255, 255);
+
 int size_dialog::idd() const {
-	return IDD_SIZE; 
+	return IDD_SIZE;
 }
 
-bool size_dialog::on_init_dialog(){
+bool size_dialog::on_init_dialog() {
+	set_int(IDC_EDIT1, board_width);
+	set_int(IDC_EDIT2, board_height);
 	return true;
 }
 
-bool size_dialog::on_ok(){
-	return true;
+bool size_dialog::on_ok() {
+	try {
+		int new_width = get_int(IDC_EDIT1);
+		int new_height = get_int(IDC_EDIT2);
+
+		if (new_width > 0 && new_height > 0 && new_width <= 20 && new_height <= 20) {
+			board_width = new_width;
+			board_height = new_height;
+			return true;
+		}
+	}
+	catch (...) {
+	}
+	return false;
 }
 
+void main_window::on_paint(HDC hdc) {
+	RECT client_rect;
+	GetClientRect(*this, &client_rect);
+
+	int cell_width = (client_rect.right - client_rect.left) / board_width;
+	int cell_height = (client_rect.bottom - client_rect.top) / board_height;
+
+	for (int row = 0; row < board_height; row++) {
+		for (int col = 0; col < board_width; col++) {
+
+			COLORREF current_color = ((row + col) % 2 == 0) ? color1 : color2;
+
+			HBRUSH brush = CreateSolidBrush(current_color);

```

Comment:
Zašto biste brush(eve) radili u petlji, a ne jednom prije petlje? `FillRect` ionako prima _brush_, ne morate ništa ni selektirati u DC.